akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/RoutersSpec.scala

```diff
@@ -115,6 +113,42 @@ class RoutersSpec extends ScalaTestWithActorTestKit("""
       }
     }
 
+    "support broadcast" in {
+      trait Cmd
+      case object ReplyWithAck extends Cmd
+      case class BCast(cmd: Cmd) extends Cmd
+
+      def behavior(replyTo: ActorRef[AnyRef]) = Behaviors.setup[Cmd] { ctx =>
+        Behaviors.receiveMessage[Cmd] {
+          case ReplyWithAck =>
+            val reply = ctx.self.path
+            replyTo ! reply
+            Behaviors.same
+          case BCast(BCast(_)) =>
+            replyTo ! new RuntimeException("totally unacceptable")
+            Behaviors.same
```

Comment:
I still don't get why we are adding this case. This is not being used at all. What is the value of having it? What am I missing? 

(btw: PR is approved, I just want to understand it)

Reply:
nothing more than sanity...
I guess this could be taken off the table if I've defined `case class BCast(text : String) extends Command` instead.

Reply:
@eyalfa, I don't think it's worth to have this. This is testing if Scala pattern matching works. 

If you send a `BCast(BCast)` message, it will pass the predicate the same way `BCast(ReplyWithAck)` does. So, there is nothing in the newly added logic and test that would make a difference for this match. 

On the other hand, if you have it defined as `case class BCast(text : String) extends Command` and the predicate test against the value of `text`, then you can send: `BCast("pass")` and `BCost("fail")` and make some assertions on it. Though I don't think necessary. The test that matter is already in place, namely that the predicate is used when defined.